src/allocation_engine/interval_tree.rs

```diff
@@ -33,14 +37,139 @@ pub struct Constraint {
     pub policy: AllocPolicy,
 }
 
-/// A closed interval range [min, max] used to describe a
+/// A closed interval range [range_start, range_end] used to describe a
 /// memory slot that will be assigned to a device by the VMM.
 /// This structure represents the key of the Node object in
 /// the interval tree implementation.
-#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Hash, Ord)]
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Hash, Ord, Debug)]
 pub struct Range {
-    pub min: u64,
-    pub max: u64,
+    /// Lower boundary of the interval.
+    range_start: u64,
+    /// Upper boundary of the interval.
+    range_end: u64,
```

Comment:
`range_` is superfluous. You are already in the `range` definition, so all the fields are already part of that namespace. `start` and `end` should suffice.